packages/components/Notification/src/Notification.tsx

```diff
@@ -62,7 +63,8 @@ export const Notification = compose<NotificationType>({
             {title && <Slots.title>{title}</Slots.title>}
             <Slots.message style={messageStyle}>{children}</Slots.message>
           </Slots.contentContainer>
-          {action ? <Slots.action>{action}</Slots.action> : !isBar && <Slots.dismissIcon>{getDismissIconPath()}</Slots.dismissIcon>}
+          {notificationButtonType === 'action' && <Slots.action onClick={onActionPress}>{action}</Slots.action>}
+          {notificationButtonType === 'dismiss' && <Slots.action icon={dismissIconProps} />}
```

Comment:
Could we pass the notificationButtonType as a prop into NotificationButton, and let NotificationButton decide what to render inside itself? That way all we have to do is render `<Slots.action {...actionProps} >` here?